Paintroid/src/androidTest/java/org/catrobat/paintroid/test/junit/intro/WelcomeActivity_RTL_Test.java

```diff
@@ -0,0 +1,120 @@
+package org.catrobat.paintroid.test.junit.intro;
+
+import android.support.v4.view.ViewPager;
+import android.test.ActivityInstrumentationTestCase2;
+import android.view.Display;
+
+import com.robotium.solo.Solo;
+
+import org.catrobat.paintroid.MainActivity;
+import org.catrobat.paintroid.R;
+import org.catrobat.paintroid.WelcomeActivity;
+
+import java.util.Locale;
+
+
+public class WelcomeActivity_RTL_Test extends ActivityInstrumentationTestCase2<MainActivity> {
+
+    public WelcomeActivity_RTL_Test() {
+        super(MainActivity.class);
+    }
+
+    private Solo solo;
+
+    @Override
+    public void setUp() throws Exception {
+        solo = new Solo(getInstrumentation(), getActivity());
+        super.setUp();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        solo.finishOpenedActivities();
+        super.tearDown();
+    }
+
+    // make sure that your PhoneLanguage is RTL like Arabic
+    public void testSupportRTL() throws Exception {
+        solo.assertCurrentActivity("Current Activity is not MainActivity", MainActivity.class);
+        assertTrue(isRTL());
+        swipeToLeft();
+        solo.clickOnText(solo.getString(R.string.help_title));
+        solo.assertCurrentActivity("Current Activity is not WelcomeActivity", WelcomeActivity.class);
+        solo.searchText(solo.getString(R.string.next));
+        solo.searchText(solo.getString(R.string.skip));
+        ViewPager viewPager = (ViewPager) solo.getView(R.id.view_pager);
+        /*
+        layouts = new int[]{
+                R.layout.islide_welcome,             0
+                R.layout.islide_tools,               1
+                R.layout.islide_possibilities,       2
+                R.layout.islide_landscape,           3
+                R.layout.islide_getstarted};         4
+        */
+        assertEquals(0, viewPager.getCurrentItem());
+        swipeToRight();
+        assertEquals(1, viewPager.getCurrentItem());
+        swipeToRight();
+        assertEquals(2, viewPager.getCurrentItem());
+        solo.sleep(400);
+        solo.clickOnText(solo.getString(R.string.next));
+        solo.clickOnText(solo.getString(R.string.next));
+        solo.clickOnText(solo.getString(R.string.next));
+        solo.clickOnText(solo.getString(R.string.next));
+        swipeToRight();
+        assertEquals(3, viewPager.getCurrentItem());
+        swipeToRight();
+        solo.sleep(600);
+        assertEquals(4, viewPager.getCurrentItem());
+        //back to start to test nextButton
+        swipeToLeft();
+        swipeToLeft();
+        swipeToLeft();
+        swipeToLeft();
+        assertEquals(0, viewPager.getCurrentItem());
+        solo.clickOnText(solo.getString(R.string.next));
+        solo.sleep(400);
```

Comment:
Just in case you want the amount of sleeping time remains intentionally the same, not only coincidentally the same. You will save effort later when you want to raise that intentional value.